### PR TITLE
Feat/4149  async es indexing

### PR DESCRIPTION
**Pull request by asf-algoo**
See #4149.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/lib/rq/__init__.py [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+import enum
+
 import redis
 import rq
 
 from tracim_backend.config import CFG
 
 
+class RqQueueName(str, enum.Enum):
+    DEFAULT = "default"
```
Comment (inkhey) [inline backend/tracim_backend/lib/rq/__init__.py:11]: what is the role of the default queue ?

File: backend/tracim_backend/tests/library/test_elasticsearch.py [hunk 2]
```diff
@@ -33,14 +37,16 @@ def content_with_parent(content_type: str, parent_type: str) -> Content:
 
 
 def a_user() -> User:
-    return User(user_id=42, display_name="Bob the sponge")
+    return User(user_id=42, display_name="Bob the sponge", email="bob@sponge.net")
```
Comment (inkhey) [inline backend/tracim_backend/tests/library/test_elasticsearch.py:40]: :sponge:

File: tools_docker/Debian_Uwsgi/supervisord_tracim.conf [hunk 3]
```diff
@@ -40,7 +40,9 @@ environment=TRACIM_CONF_PATH=/etc/tracim/development.ini
 ; RQ database worker
 [program:tracim_rq_worker]
 directory=/tracim/backend/
-command=rq worker -q -w tracim_backend.lib.rq.worker.DatabaseWorker event
+; NOTE 2021-02-23 - S.G. queue names should stay the same as RqQueueName enum values
+; mail_sender is separate as it has its own worker (named tracim_mail_notifier, just above)
+command=rq worker -q -w tracim_backend.lib.rq.worker.DatabaseWorker event elasticsearch_indexer
```
Comment (inkhey) [inline tools_docker/Debian_Uwsgi/supervisord_tracim.conf:45]: Can you check if something can be missing in documentation `/backend/doc` about this new worker ?

Comment (grignards) [reply tools_docker/Debian_Uwsgi/supervisord_tracim.conf:45]: It's not new per se (it has been added for TLMs), a rq worker needs to be explicitly given its queue names hence the change + the note here.

File: backend/tracim_backend/lib/rq/__init__.py (hunk 1 above)
Comment (grignards) [reply backend/tracim_backend/lib/rq/__init__.py:11]: A RQ worker uses this queue by default, it was the default argument of `get_rq_queue`.

File: tools_docker/Debian_Uwsgi/supervisord_tracim.conf (hunk 3 above)
Comment (grignards) [reply tools_docker/Debian_Uwsgi/supervisord_tracim.conf:45]: But you're right, a part of the documentation is explaining how to launch this worker, I'll edit it.

File: backend/tracim_backend/lib/rq/__init__.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/rq/__init__.py:11]: ok. understood.

File: tools_docker/Debian_Uwsgi/supervisord_tracim.conf (hunk 3 above)
Comment (grignards) [reply tools_docker/Debian_Uwsgi/supervisord_tracim.conf:45]: Done.

**Review by inkhey — APPROVED** [review]

**Review by raphj — APPROVED** [review]

**Comment by grignards** [comment]
Tested on branch, by hand and with the docker image.

**Pull request merged**